src/sqsd.groovy

```diff
@@ -68,8 +68,8 @@ try {
         def messages = sqs.receiveMessage(receiveMessageRequest).getMessages()
         println "Received Messages : " + messages.size()
 
-        // Break when empty
-        if(messages.size() <= 0) break
+        // Break when empty if not running daemonized
+        if(messages.size() <= 0 && config.sqsd.run_daemonized < 1) break
```

Comment:
I am failing to see how it stays deaemonized using this code. As @abdielou says, it should sleep (Thread.sleep) for some configurable time when daemonized. However, I agree with @jeffcharles in that it interferes with the correct functioning of the long poll, but don't see another easy way in making it work as a daemon. Although not documented here, I have thought about this and the solution I think would be best in your case is to use a cron tab inside the docker container, not outside. Then I would configure the cron expression in a system property so that when the Docker image boots, it uses the specified expression to run the `sqsd` permanently. In that way the `sqsd` normal workflow is not altered and the long polling mechanism will behave as it is supposed.